src/app/components/home.component.ts

```diff
@@ -1682,6 +1695,10 @@ export class HomeComponent implements OnInit, AfterViewInit {
       if (settingsObject.buttonSettings[element]) {
         this.settingsButtons[element].toggled = settingsObject.buttonSettings[element].toggled;
         this.settingsButtons[element].hidden = settingsObject.buttonSettings[element].hidden;
+        // retrieving state of buttons for touchBar
+        if (this.settingsButtons[element].toggled) {
```

Comment:
This should include `codeRunningOnMac` and probably a `touchBarEnabled` variable - see #312